### PR TITLE
docs(recurring job): behavior changed

### PR DESCRIPTION
**Pull request by mantissahz**
Update `important-notes` for v1.3.3, v1.4.2 and v1.5.0.

Ref: longhorn/longhorn#4898

### CODE REVIEW

**Comment by mantissahz** [comment]
cc @c3y1huang @innobead

File: content/docs/1.3.3/deploy/important-notes/index.md [hunk 1]
```diff
@@ -14,3 +14,4 @@ Please see [here](https://github.com/longhorn/longhorn/releases/tag/v{{< current
 4. There are two important fixes that will prevent rarely potential data corruption during replica rebuilding, and also improve write performance via the specific filesystem block size.
    - [4354](https://github.com/longhorn/longhorn/issues/4354): Introduce Data Alignment Correction for existing volumes if the filesystem block size is less than 4096.
    - [4594](https://github.com/longhorn/longhorn/issues/4594): Use the specific block size for the filesystem to avoid unnecessary Ready-Modify-Write operations between volume head and snapshots.
+5. After the upgrade, the behavior of recurring job task types `Snapshot` and `Backup` will be changed to check the number of old snapshots first. If the number of old snapshots exceeds the retain count, older snapshots will be cleaned up first before taking a new snapshot. And introduce two new recurring job task types `Snapshot Force Create` and `Backup Force Create` and the new task type `Snapshot Force Create` will behave the same as the original behavior of the task type `Snapshot` as well as task types `Backup Force Create` and `Backup`.
```
Comment (innobead) [inline content/docs/1.3.3/deploy/important-notes/index.md:17]:
```suggestion
5. After the upgrade, the behavior of the recurring job types `Snapshot` and `Backup` will attempt to delete old snapshots first if they exceed the retained count before creating a new snapshot. Additionally, two new recurring job types have been introduced, `Snapshot Force Create` and `Backup Force Create`. They retain the original behavior of taking a snapshot or backup first before deleting outdated snapshots.
```

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md [hunk 2]
```diff
@@ -65,8 +65,10 @@ The following parameters should be specified for each recurring job selector:
 - `name`: Name of the recurring job. Do not use duplicate names. And the length of `name` should be no more than 40 characters.
 
 - `task`: Type of the job. Longhorn supports the following:
-  - `backup`: periodically create snapshots then do backups
-  - `snapshot`: periodically create snapshots
+  - `backup`: periodically create snapshots then do backups except for old snapshots cleanup failed before creating new snapshots
```
Comment (innobead) [inline content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md:68]:
```suggestion
  - `backup`: periodically create snapshots then do backups after cleaning up outdated snapshots
```

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md [hunk 3]
```diff
@@ -65,8 +65,10 @@ The following parameters should be specified for each recurring job selector:
...
+  - `backup-force-create`: periodically create snapshots the do backups even if old snapshots cleanup failed
+  - `snapshot`: periodically create snapshots except for old snapshots cleanup failed before creating new snapshots
```
Comment (innobead) [inline content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md:70]:
```suggestion
  - `snapshot`: periodically create snapshots after cleaning up outdated snapshots
```

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md [hunk 4]
```diff
@@ -65,8 +65,10 @@ The following parameters should be specified for each recurring job selector:
...
+  - `snapshot-force-create`: periodically create snapshots even if old snapshots cleanup failed
```
Comment (innobead) [inline content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md:71]:
```suggestion
  - `snapshot-force-create`: periodically create snapshots
```

File: content/docs/1.4.2/deploy/important-notes/index.md [hunk 5]
```diff
@@ -14,6 +14,8 @@ Please ensure your Kubernetes cluster is at least v1.21 before upgrading to Long
 ### Recurring Jobs
 After the upgrade, the recurring job settings of volumes will be migrated to new recurring job resources, and the `RecurringJobs` field in the volume spec will be deprecated. [[doc](https://longhorn.io/docs/{{< current-version >}}/deploy/upgrade/#4-automatically-migrate-recurring-jobs)]
 
+And the behavior of recurring job task types `Snapshot` and `Backup` will be changed to check the number of old snapshots first. If the number of old snapshots exceeds the retain count, older snapshots will be cleaned up first before taking a new snapshot. And introduce two new recurring job task types `Snapshot Force Create` and `Backup Force Create` and the new task type `Snapshot Force Create` will behave the same as the original behavior of the task type `Snapshot` as well as task types `Backup Force Create` and `Backup`.
```
Comment (innobead) [inline content/docs/1.4.2/deploy/important-notes/index.md:17]:
```suggestion
The behavior of the recurring job types `Snapshot` and `Backup` will attempt to delete old snapshots first if they exceed the retained count before creating a new snapshot. Additionally, two new recurring job types have been introduced, `Snapshot Force Create` and `Backup Force Create`. They retain the original behavior of taking a snapshot or backup first before deleting outdated snapshots.
```

File: content/docs/1.5.0/deploy/important-notes/index.md [hunk 6]
```diff
@@ -14,6 +14,8 @@ Please ensure your Kubernetes cluster is at least v1.21 before upgrading to Long
 ### Recurring Jobs
 After the upgrade, the recurring job settings of volumes will be migrated to new recurring job resources, and the `RecurringJobs` field in the volume spec will be deprecated. [[doc](https://longhorn.io/docs/{{< current-version >}}/deploy/upgrade/#4-automatically-migrate-recurring-jobs)]
 
+And the behavior of recurring job task types `Snapshot` and `Backup` will be changed to check the number of old snapshots first. If the number of old snapshots exceeds the retain count, older snapshots will be cleaned up first before taking a new snapshot. And introduce two new recurring job task types `Snapshot Force Create` and `Backup Force Create` and the new task type `Snapshot Force Create` will behave the same as the original behavior of the task type `Snapshot` as well as task types `Backup Force Create` and `Backup`.
```
Comment (innobead) [inline content/docs/1.5.0/deploy/important-notes/index.md:17]:
```suggestion
The behavior of the recurring job types `Snapshot` and `Backup` will attempt to delete old snapshots first if they exceed the retained count before creating a new snapshot. Additionally, two new recurring job types have been introduced, `Snapshot Force Create` and `Backup Force Create`. They retain the original behavior of taking a snapshot or backup first before deleting outdated snapshots.
```

**Review by innobead — CHANGES_REQUESTED** [review]
@mantissahz Please correct the doc as per the suggestions.

I did not make all the suggested changes, so you need to recheck this PR and fix them all.

File: content/docs/1.3.3/deploy/important-notes/index.md (hunk 1 above)
Comment (mantissahz) [reply content/docs/1.3.3/deploy/important-notes/index.md]: Modified. Thanks

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md (hunk 2 above)
Comment (mantissahz) [reply content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md]: Modified for all related files. Thanks

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md (hunk 3 above)
Comment (mantissahz) [reply content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md]: Modified for all related files. Thanks

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md (hunk 4 above)
Comment (mantissahz) [reply content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md]: Modified for all related files. Thanks

File: content/docs/1.4.2/deploy/important-notes/index.md (hunk 5 above)
Comment (mantissahz) [reply content/docs/1.4.2/deploy/important-notes/index.md]: Modified. Thanks

File: content/docs/1.5.0/deploy/important-notes/index.md (hunk 6 above)
Comment (mantissahz) [reply content/docs/1.5.0/deploy/important-notes/index.md]: Modified. Thanks

File: content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md (hunk 2 above)
Comment (mantissahz) [reply content/docs/1.3.3/snapshots-and-backups/scheduling-backups-and-snapshots.md]: Modified for all related files. Thanks

File: content/docs/1.5.0/deploy/important-notes/index.md (hunk 6 above)
Comment (mantissahz) [reply content/docs/1.5.0/deploy/important-notes/index.md]: Modified. Thanks

**Review by innobead — APPROVED** [review]
LGTM.

**Pull request merged**